reconcile/templating/renderer.py

```diff
@@ -107,6 +106,31 @@ def read(self, path: str) -> Optional[str]:
         return current
 
 
+class PersistenceContext(FilePersistence):
+    def __init__(self, persistence: FilePersistence, dry_run: bool) -> None:
```

Comment:
A little strange to both inherit `FilePersistence` and pass in another `FilePersistence`, maybe call it something like `TransactionPersistence` and add some docstring?

Reply:
Yeah, `TransactionablePersistence` was my first take as well. I can rename it.